pkg/streamingpromql/testdata/ours/subqueries.test

```diff
@@ -0,0 +1,79 @@
+# SPDX-License-Identifier: AGPL-3.0-only
+
+# Most cases for functions are covered already in the upstream test cases.
+# These test cases cover scenarios not covered by the upstream test cases, such as range queries, or edge cases that are uniquely likely to cause issues in the streaming engine.
+
+load 1m
+  metric 0 4 3 6 -1 10
```

Comment:
We should include some tests with histograms to make sure that buffer is handled correctly. Maybe a second series with histograms, and a third with both types?

Reply:
Added in ac84802.